File: DPP/21-06-2026/Python/Q2.py
from typing import List

class Solution:
    def numberOfSmallerPairs(self, nums: List[int], target: int) -> int:
        count = 0
        left = 0
        right = len(nums) - 1

        while left < right:
            currentSum = nums[left] + nums[right]
            
            if currentSum < target:
                count += right - left
                left += 1
            else:
                right -= 1
        
        return count
    
sol = Solution()
nums = [1, 2, 3, 4]
target = 6
result = sol.numberOfSmallerPairs(nums, target)
print(result)

File: DPP/21-06-2026/Java/Q2.java
import java.util.Arrays;

public class Q2 {
    public static void main(String[] args) {
        Solution sol = new Solution();
        int[] nums = {1, 2, 3, 4};
        int target = 6;
        int result = sol.numberOfSmallerPairs(nums, target);

        System.out.println(result);
    }
}

class Solution {
    public int numberOfSmallerPairs(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;

        int count = 0;

        while (left < right) {
            int currentSum = nums[left] + nums[right];
            
            if (currentSum < target) {
                count += (right - left);
                left += 1;
            } else {
                right -= 1;
            }
        }

        return count;
    }
}